src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -114,7 +115,7 @@ public static Type fromId(byte id) throws ElasticsearchIllegalArgumentException
         }
     }
 
-    private volatile Stage stage = Stage.INIT;
+    private volatile Stage stage;
 
```

Comment:
OK. will do. I added the sync on the write method to avoid concurrent writes (because of the check). I also added syncronized to the readFrom method.